### PR TITLE
Fix two mistakes and refactor code for daily xp behaviour.

### PR DESCRIPTION
**Pull request by shomyx-dev**
Fix hasBoth = combatTips && combatTips. 
Fix missing multiplier for raiseTheMeek. 

And a rather personal opinion:
Refactor and cleanup code.

### CODE REVIEW

**Comment by shomyx-dev** [comment]
Also if you could clarify in config.xml the following:

`
	<ScaleByReadyToUpgrade>false</ScaleByReadyToUpgrade>`

Because it's not clear which value for the setting does what from the comment, and by looking at the code it sould seem that if it is **true**, then it has the intended effect.
I would add "if true" in the beginning as such:
`<!-- If true, XP for any ready to upgrade troops is NOT given to the stack -->`

File: src/Behaviours/MobilePartyDailyTickBehaviour.cs [hunk 1]
```diff
@@ -16,49 +16,48 @@ public override void SyncData(IDataStore dataStore)
...
-            {
-                hasBoth = true;
-            }
-            if (party.IsActive && party.HasPerk(DefaultPerks.Leadership.CombatTips))
+            if (!party.IsActive)
+                return;
+
+            bool hasCombatTips = party.HasPerk(DefaultPerks.Leadership.CombatTips);
+            bool hasRaiseTheMeek = party.HasPerk(DefaultPerks.Leadership.RaiseTheMeek);
+
```
Comment (shomyx-dev) [inline src/Behaviours/MobilePartyDailyTickBehaviour.cs:24]: Could also return early if both are false to avoid a useless enumeration.

**Comment by tynakuh** [comment]
Thanks, appreciate this, was definitely a mess before, especially now that I read it back 0.o

**Comment by tynakuh** [comment]
Agreed on clarification on the config, feel free to improve the wording in this PR if you'd like, else i'll push those up to master when I'm back from work along with verifying v1.1.0 (Should be fine.)

I'll dig through the other code and see if they require a refactor too, much of it was rather rushed/done while tired.

**Comment by shomyx-dev** [comment]
Sure, I'll add some wording modifications.

Do note that in 1.1.0 patchnotes it says
> Troops can now upgrade more easily.

If you have time and could make sure that the multiplier issue (xp being applied once per stack instead of once per unit in stack) is still present (and clarify this on nexusmods & here too) it would be great.

**Comment by tynakuh** [comment]
That's the plan 👍

**Pull request merged**